app/web_ui/src/routes/(app)/docs/rag_configs/[project_id]/run_rag_control.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="ts">
+  import Dialog from "$lib/ui/dialog.svelte"
+  import type { RagConfigWithSubConfigs } from "$lib/types"
+  import RunRagDialog from "./run_rag_dialog.svelte"
+  import { ragProgressStore } from "$lib/stores/rag_progress_store"
+  import type { RagConfigurationStatus } from "$lib/stores/rag_progress_store"
+
+  export let btn_size: "normal" | "mid" = "mid"
+  export let project_id: string
+  export let rag_config: RagConfigWithSubConfigs
+
+  $: rag_config_id = rag_config.id || ""
+
+  let run_rag_dialog: Dialog | null = null
+
+  function get_state_to_label(status: RagConfigurationStatus) {
+    const state_to_label: Record<
+      string,
+      {
+        label: string
+        icon: "play" | "loading" | null
+        action: () => void
+        isPrimary: boolean
+      } | null
+    > = {
+      not_started: {
+        label: "Run",
+        icon: "play",
+        isPrimary: true,
+        action: () => {
+          run_rag_dialog?.show()
+        },
+      },
+      running: {
+        label: "Running",
+        icon: "loading",
+        isPrimary: false,
+        action: () => {
+          run_rag_dialog?.show()
+        },
+      },
+      completed_with_errors: {
+        label: "View details",
+        icon: null,
+        isPrimary: false,
+        action: () => {
+          run_rag_dialog?.show()
+        },
+      },
+      complete: {
+        label: "View details",
```

Comment:
why even have this be clickable for success? The screen you bring up doesn't have extra information you can't already see. We can already see the count, and that everything is 100%. It's useful for running/errors, but not success.

See comment about merging with the badge before working on this alone.